chore(aws): eks targeted helm guide to deploy C8

Langleu opened the pull request:
## Description

<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
<!-- Add an assignee and use component: labels for good hygiene! -->

This PR is just preliminary and targeting a separate branch, not main! It's just a PR to get a technical review.
This does not follow writing best practices and will be reworked in the future when creating a PR from INFEX-6 to main.

For the reviewer, switched to `mdx` to profit from some additional feature.

DevEx / Christina does not require a review from your side atm, but thanks 🙇!

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,208 @@
+  </TabItem>
+</Tabs>
+
+With the C8 Helm chart deployed, consider continuing your journey by [interacting with the zeebe-gateway](https://docs.camunda.io/docs/self-managed/zeebe-deployment/security/client-authorization/).
```
nice automation 🎉

akeller replied:
☝️ Some kudos for you @pepopowitz

maxdanilov commented:
Comments as discussed:
- Add the page to the contents on the left
- Pin the Helm chart versions
- Have a closing part to the guide: how to check all C8 components from the chart are running, how to work with the C8 apps (e.g. how to open Operate UI)

Langleu commented:
- Added the page to the sidebar for easier access
- dropped the patch versions
- pinned the helm charts
- have a "closing chapter" to verify that C8 works by either utilizing zbctl or the modeler (with / without domain)
- rest are just references to our stack components as those contain guides on how to use / interact with them.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+export DOMAIN_NAME=camunda.example.com
+# The e-mail to register with Let's Encrypt
+export MAIL=admin@camunda.example.com
+# The Ingress Helm Chart version
```
Let's explicitly specify which one we use:
```suggestion
# The Ingress-Nginx Helm Chart version
export INGRESS_NGINX_HELM_CHART_VERSION="4.8.3"
```

maxdanilov replied:
let's also quote the version so Docusaurus highlight is not confused (see the patch versions off-color):

<img width="411" alt="Screenshot 2023-11-29 at 15 46 31" src="https://github.com/camunda/camunda-docs/assets/6655714/d805bec7-abf5-418d-8442-416e171c20b8">

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+
+#### ingress-nginx
+
+Ingress-nginx is an open-source Kubernetes Ingress controller that provides a way to manage external access to services within a Kubernetes cluster. It acts as a reverse proxy and load balancer, routing incoming traffic to the appropriate services based on rules defined in the Ingress resource.
```
let's add a link to the project so people are 100% aligned on which one we use: https://github.com/kubernetes/ingress-nginx

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+
+Ingress-nginx is an open-source Kubernetes Ingress controller that provides a way to manage external access to services within a Kubernetes cluster. It acts as a reverse proxy and load balancer, routing incoming traffic to the appropriate services based on rules defined in the Ingress resource.
+
+The following will install `ingress-nginx` in the `ingress-nginx` namespace via Helm. For more configuration options, conduct the [Helm chart](https://github.com/kubernetes/ingress-nginx/tree/main/charts/ingress-nginx).
```
```suggestion
The following will install `ingress-nginx` in the `ingress-nginx` namespace via Helm. For more configuration options, consult the [Helm chart](https://github.com/kubernetes/ingress-nginx/tree/main/charts/ingress-nginx).
```

maxdanilov replied:
same for the other usages of `conduct` I assume.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+
+#### cert-manager
+
+Cert-manager is an open-source Kubernetes add-on that automates the management and issuance of TLS certificates. It integrates with various certificate authorities (CAs) and provides a straightforward way to obtain, renew, and manage SSL/TLS certificates for your Kubernetes applications.
```
Same here, let's link to the project: https://cert-manager.io/

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+
+#### external-dns
+
+External-dns is a Kubernetes add-on that automates the management of DNS records for external resources, such as Load Balancers or Ingress controllers. It monitors the Kubernetes resources and dynamically updates the DNS provider with the appropriate DNS records.
```
Let's link to the project explicitly: https://github.com/kubernetes-sigs/external-dns

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
````diff
@@ -0,0 +1,360 @@
+  --set ingressShim.defaultIssuerGroup=cert-manager.io
+```
+
+It is recommended to install the cert-manager custom resource definitions (CRDs) separately from Helm. Please refer to the [installation advice](https://cert-manager.io/docs/installation/helm/#crd-installation-advice) provided by cert-manager. This approach is recommended because when using Helm, there is a possibility of accidentally omitting the CRDs due to incorrect configuration resulting the ...
````
In such case, I'd say we either need to modify the `helm install` command (to install CRDs separately from the app itself) or drop this recommendation, otherwise we're providing an example that we call unrecommended the very next paragraph.

Langleu replied:
will have a look, maybe do the "recommended" way but we also mention not production ready and providing links to further readings, so I think it kinda complies with the expectations.

Langleu replied:
separated the two by first installing the crds via kubectl and then installing the chart.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm.mdx:
```diff
@@ -0,0 +1,360 @@
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-production
```
maybe drop `production`? This is the first time we have anything about prod mentioned and I think it may be confusing:
```suggestion
  name: letsencrypt
```

maxdanilov reviewed (changes requested):
Very nice, I especially like how it's simply explained how to use Identity/Keycloak creds.

Leving some minor comments for clarification.

maxdanilov reviewed (approved):
🚀
